.circleci/config.yml

```diff
@@ -0,0 +1,41 @@
+
+build: &build
+  docker:
+    - image: majorz/rust-$CIRCLE_JOB:rust-1.23.0
+  steps:
+    - checkout
+    - run: cargo build
+
+version: 2
+
+jobs:
+  aarch64:
+    <<: *build
+  rpi:
+    <<: *build
+  armv7hf:
+    <<: *build
+  i386:
+    <<: *build
+  amd64:
+    <<: *build
+  test:
+    docker:
+      - image: majorz/rust-test-deploy:rust-nightly-2018-01-08
```

Comment:
Is it a good idea to pin this to a single nightly version?

Reply:
My idea is to bump the nightly version in this place once new image version is published. The commit will go hand in hand with the needed clippy and rustfmt updates. This will solve the problem we often have where a new PR is created and it gets blocked by the fact that clippy has a new version and the build does not pass because of it. So a second PR has to be created, then a rebase is needed, etc. I think it is better to manually bump it up once in a while on Rust stable releases and important clippy versions.

Reply:
Ah, one important note is that the Rust nightly version is the one that the clippy version from the image targets, e.g.:
https://github.com/rust-lang-nursery/rust-clippy/blob/master/CHANGELOG.md#00178
https://github.com/resin-io-projects/rust-cross-toolchains/blob/d1b0c745b8762d3bc0b42d116510cabc80a02693/.circleci/config.yml#L47

Sometimes with the current approach of using latest nightly + latest clippy, latest clippy does not compile with latest nightly, so the PR gets blocked awaiting for a new clippy release.

Reply:
Gotcha.